pkg/sbom/v1/sbom_spdx_storage_format.go

```diff
@@ -163,13 +188,28 @@ func (sbom *SBOMData) FilterSBOM(sbomFileRelevantMap map[string]bool) error {
 		}
 	}
 
+	//filter relevant file list from package source Info
+	relevantPackageFromSourceInfoMap := make(map[spdxv1beta1.ElementID]bool)
+	for realtimeFileName := range sbomFileRelevantMap {
+		if data, _ := sbom.relevantRealtimeFilesByPackageSourceInfo.Load(realtimeFileName); data != nil && !data.(*packageSourceInfoData).exist {
```

Comment:
`!data.(*packageSourceInfoData).exist` This is not good - What if the parsing fails? 
You should have here a separated `if` statement